packages/sit-onyx/src/components/OnyxBadge/OnyxBadge.vue

```diff
@@ -0,0 +1,106 @@
+<script lang="ts" setup>
+import { useDensity } from "../../composables/density";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import type { OnyxBadgeProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxBadgeProps>(), {
+  variation: "primary",
+});
+
+const { densityClass } = useDensity(props);
+
+defineSlots<{
+  /**
+   * Badge content.
+   */
+  default(): unknown;
```

Comment:
```suggestion
  default?(): unknown;
```

The slot is optional if only an icon is used, right?

Reply:
hmm, yes. If an icon is set I just ignore the slot. No harm in setting it optional